### PR TITLE
First draft ISA95 test

### PR DESCRIPTION
**Pull request by huguesBouvier**
Add tests in ISA95 environment.
At the moment those tests only run ping and temp tests (validate up and down) because the tests are not running on lvl3 but are running on "jumpbox" a separate VM with internet access.

This PR also adds the ability for lvl 5 and lvl 3 to go through a proxy.

Additionally, this PR adds Nested E2E test with lvl4 and lvl5 deployed with AMQP.

It also fixes an instability in the temp sensor test

AMQP: https://msazure.visualstudio.com/One/_build/results?buildId=39261869&view=logs&j=e8afecc0-c259-5cc1-62e0-5c5d1c2a3db4&t=a5111985-4f75-51bb-4ed8-1624e65f53ff
ISA: https://dev.azure.com/msazure/One/_build/results?buildId=39246416&view=logs&j=c4ceec0d-93ba-54ca-fe7b-053bb822a9ca
Regression E2E: https://dev.azure.com/msazure/One/_build/results?buildId=39248358&view=logs&j=b9201419-b8e8-5f70-036b-4feb1c7a3361&t=db23c1c7-adbf-5885-c7dc-2d3f5c08068b

### CODE REVIEW

File: builds/e2e/templates/e2e-run.yaml [hunk 1]
```diff
@@ -18,22 +20,30 @@ steps:
       $filter += '&Category!=UnstableOnArm'
     }
     
-    #filter out tests not usable on nested edge
-    if ('$(nestededge)' -eq 'true')
+    if ($test_type -eq 'nestededge_mqtt')
     {
       $filter += '&Category!=SingleNodeOnly'
```
Comment (richma-ms) [inline builds/e2e/templates/e2e-run.yaml:25]: >$filter += '&Category!=SingleNodeOnly' [](start = 6, length = 38)

I think this needs to apply to nestededge_amqp as well

File: test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs [hunk 2]
```diff
@@ -91,7 +91,8 @@ IEnumerable<Registry> GetAndValidateRegistries()
             this.MethodSenderImage = Option.Maybe(Get("methodSenderImage"));
             this.OptimizeForPerformance = context.GetValue("optimizeForPerformance", true);
             this.PackagePath = Option.Maybe(Get("packagePath"));
-            this.Proxy = Option.Maybe(context.GetValue<Uri>("proxy"));
+            this.AgentProxy = Option.Maybe(context.GetValue<Uri>("agentProxy"));
```
Comment (richma-ms) [inline test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs:94]: >AgentProxy [](start = 17, length = 10)

maybe TestRunnerProxy? I stared at this for a good 30sec trying to understand why the EdgeAgent needed a different proxy :D

File: builds/e2e/isa-95-smoke-test.yaml [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+  - job: SetupVM_level5
+    displayName: SettingUp level 5
+    timeoutInMinutes: 180
+    variables:
+      artifactName: iotedged-ubuntu18.04-amd64
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb
+      vsts.project: $(System.TeamProjectId)
+      level: '5'
+      deploymentFile: 'nestededge_topLayerBaseDeployment.json'
```
Comment (richma-ms) [inline builds/e2e/isa-95-smoke-test.yaml:30]: >'nestededge_topLayerBaseDeployment.json' [](start = 22, length = 40)

shouldn't this be _amqp?  I thought golden path was suppose to be without broker.

File: builds/e2e/isa-95-smoke-test.yaml [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    condition: succeeded()
+    timeoutInMinutes: 180
+    variables:
+      identityServiceArtifactName: packages_ubuntu-18.04_amd64
+      identityServicePackageFilter: aziot-identity-service_*_amd64.deb   
+      artifactName: iotedged-ubuntu18.04-amd64
+      vsts.project: $(System.TeamProjectId)
+      parentName: $[ dependencies.SetupVM_level5.outputs['deployIoTEdge.deviceName'] ]
+      parentDeviceId: $[ dependencies.SetupVM_level5.outputs['createIdentity.parentDeviceId'] ] 
+      deploymentFile: 'nestededge_middleLayerBaseDeployment.json'
```
Comment (richma-ms) [inline builds/e2e/isa-95-smoke-test.yaml:67]: >'nestededge_middleLayerBaseDeployment.json' [](start = 22, length = 43)

same here

File: e2e_deployment_files/nestededge_isa95_smoke_test_BaseDeployment.json [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            "env": {
+              "experimentalFeatures__enabled": {
+                "value": "true"
+              },
+              "experimentalFeatures__nestedEdgeEnabled": {
+                "value": "true"
+              },
+              "experimentalFeatures__mqttBrokerEnabled": {
+                "value": "true"
+              },
```
Comment (richma-ms) [inline e2e_deployment_files/nestededge_isa95_smoke_test_BaseDeployment.json:39]: should the broker be turned on for L3?

File: test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs (hunk 2 above)
Comment (huguesBouvier) [reply test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs:94]: Fixed

File: builds/e2e/templates/e2e-run.yaml (hunk 1 above)
Comment (huguesBouvier) [reply builds/e2e/templates/e2e-run.yaml:25]: Good catch

File: builds/e2e/isa-95-smoke-test.yaml (hunk 3 above)
Comment (huguesBouvier) [reply builds/e2e/isa-95-smoke-test.yaml]: Fixed

File: e2e_deployment_files/nestededge_isa95_smoke_test_BaseDeployment.json (hunk 5 above)
Comment (huguesBouvier) [reply e2e_deployment_files/nestededge_isa95_smoke_test_BaseDeployment.json]: Fixed

File: builds/e2e/isa-95-smoke-test.yaml (hunk 4 above)
Comment (huguesBouvier) [reply builds/e2e/isa-95-smoke-test.yaml]: Fixed

**Review by richma-ms — APPROVED** [review]

File: builds/e2e/templates/e2e-run.yaml [hunk 6]
```diff
@@ -18,22 +20,31 @@ steps:
...
     }
+    else
+    {
+      $filter += '&Category!=NestedEdgeOnly' 
+    } 
 
     sudo --preserve-env dotnet test $testFile --logger:trx --testcasefilter:$filter  
-  displayName: Run tests
+  displayName: Run tests ${{ parameters.test_type }}
+  continueOnError: true
```
Comment (vadim-kovalyov) [inline builds/e2e/templates/e2e-run.yaml:47]: @huguesBouvier can I ask why this change? it makes it harder to re-run failed tests in the pipeline. If you want some other steps to still execute even if tests are failed, you can use `condition: always` instead.
